analyses/ewings-cell-type/01-marker-gene-tumor-classification.Rmd

```diff
@@ -0,0 +1,516 @@
+CD99 appears to be expressed in both normal and tumor at similar levels. 
+All other genes show differences between the two, although at varying degrees. 
+Again, we should make sure this is consistent across multiple patients before removing these genes as marker genes. 
+
+### Marker genes using `findMarkers`
+
+Below, we will use `scran::findMarkers()` to find any markers between tumor and normal cells. 
+Since we used marker genes to classify tumor cells, I would expect that at least some of the top markers would be from our list of tumor markers. 
+
+```{r}
+# add in tumor/normal classification to colData of SCE 
+celltypes_only_df <- marker_sum_exp |> 
+  dplyr::select(barcodes, sum_classification)
+
+coldata_df <- as.data.frame(colData(sce)) |> 
+  dplyr::left_join(celltypes_only_df)
+
+colData(sce) <- DataFrame(coldata_df, row.names = rownames(coldata_df))
+
+# find markers between tumor and normal 
+markers <- scran::findMarkers(sce,
+                              groups = sce$sum_classification,
+                              pval.type = "all") 
+```
+
+```{r}
+# pull out top markers for tumor cells 
+tumor_markers <- markers$Tumor |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("ensembl_gene_id") |> 
+  dplyr::left_join(marker_genes_df) |>
+  # just take those with FDR < 0.05 and logFC > 1.5
+  dplyr::filter(FDR < 0.05, summary.logFC > 1.5) |> 
+  dplyr::arrange(FDR) 
+
+# get any marker genes that are in both findMarkers and our curated list
+marker_gene_intersect <- tumor_markers |> 
+  dplyr::filter(cell_type == "tumor")
+
+# print out genes that are marker genes for both 
+marker_gene_intersect
+```
+
+### Using clusters and marker gene expression to annotate tumor cells
+
+We anticipate that the tumor cells and normal cells would cluster separately. 
+First, let's just plot the cluster assignments alone.  
+
+These are clusters calculated in `scpca-nf` and its probable that these aren't the best clusters to use. 
+Separately, we will evaluate clustering for these samples, but for now we will look at the clusters we have. 
+
+```{r}
+# plot clusters on UMAP
+ggplot(marker_sum_exp, aes(x = UMAP1, y = UMAP2, color = cluster)) +
+  geom_point(size = 0.5, alpha = 0.5) +
+  theme_bw()
+```
+
+It looks like our large group of cells on the bottom right is comprised of a few clusters while most other groups of cells are a single cluster. 
+
+We can then look at marker gene expression across clusters.
+Below we visualize the total number of cells in each cluster that are classified as tumor or normal using the sum of marker gene expression. 
+Then for each set of cells, how many of those actually express each marker gene. 
+
+We expect to see two things: 
+
+- Tumor cells and normal cells tend to separate by cluster. 
+- Good marker genes will be expressed in tumor cells but not normal cells. 
+
+```{r, fig.height=10, fig.width=10}
```

Comment:
I don't think I have a good suggestion here, but I don't love this plot. My main complaint is mostly that the counts are so small for some of the clusters that it is hard to see the values. But transforming doesn't necessarily help, and using free scales for the facets would be very busy. It may be that this is just too much for one plot, but as this is an exploration notebook, it is probably fine to leave as is.

Reply:
The reason I made this plot is because I wanted to see how many of each cell type actually express the marker gene above our threshold of 0. I think this helps show which marker genes are going to be more specific to tumor cells, although not a complete analysis at all, I do think it's helpful to see. It was very busy though so I updated this plot to only separate by tumor/normal and not include the clusters. Hopefully it's a little bit easier to interpret this way.